src/main/java/it/smartcommunitylab/aac/webauthn/controller/WebAuthnRegistrationController.java

```diff
@@ -0,0 +1,195 @@
+    @RequestMapping(value = "/auth/webauthn/register/{providerId}", method = RequestMethod.GET)
+    public String registrationPage(@PathVariable("providerId") String providerId) {
+        try {
+            final boolean canRegister = registrationRepository.findByProviderId(providerId)
+                    .getConfigMap()
+                    .isEnableRegistration();
+            if (!canRegister) {
+                throw new RegistrationException("registration is disabled");
+            }
+            return "webauthn/register";
+        } catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Starts a new WebAuthn registration ceremony by generating a new Credential
+     * Creation Options object and returning it to the user.
+     * 
+     * The challenge and the information about the ceremony are temporarily stored
+     * in the session.
+     */
+    @Hidden
+    @PostMapping(value = "/auth/webauthn/attestationOptions/{providerId}", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public WebAuthnRegistrationResponse generateAttestationOptions(
+            @RequestBody @Valid WebAuthnRegistrationStartRequest body,
+            @PathVariable("providerId") String providerId) {
+        try {
+            final boolean canRegister = registrationRepository.findByProviderId(providerId)
+                    .getConfigMap()
+                    .isEnableRegistration();
+            if (!canRegister) {
+                throw new RegistrationException("registration is disabled");
+            }
+            WebAuthnRpService rps = webAuthnRpServiceReigistrationRepository.get(providerId);
+            String username = body.getUsername();
+            if (!isValidUsername(username)) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid username");
+            }
+
+            String displayName = body.getDisplayName();
+            if (!isValidDisplayName(displayName)) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid displayName");
+            }
+
+            final WebAuthnIdentityService webAuthnIdentityService = webAuthnIdentityAuthority
+                    .getIdentityService(providerId);
+            final WebAuthnSubjectResolver subjectResolver = webAuthnIdentityService.getSubjectResolver();
+            Subject subject = subjectResolver
+                    .resolveByUserId(username);
+            WebAuthnIdentityProviderConfig providerCfg = registrationRepository.findByProviderId(providerId);
+            String realm = providerCfg.getRealm();
+            final WebAuthnRegistrationResponse response = rps.startRegistration(
+                    username,
+                    realm,
+                    displayName,
+                    subject);
+            return response;
+        } catch (Exception e) {
```

Comment:
remove the try/catch

Reply:
Addressed with commits 45607818fb009ed11fd571f703dbaf550a847db8 and 2702bb2c1d53e5f9c6f6e50cd1095602cf08b997